src/components/proposal/ProposalContainer.tsx

```diff
@@ -150,17 +151,37 @@ export default function ProposalContainer(props: {
     }
   }, [previousInitialProposal, props.proposal, dispatch]);
 
+  const hasReferenceNumberFormat = state.proposal.call?.referenceNumberFormat
+    ? true
+    : false;
+
+  const isSubmitted = state.proposal.submitted;
+
+  const proposalId = state.proposal.proposalId;
```

Comment:
Here also something like: `const { isSubmitted, proposalId } = state.proposal;` will clean up a bit so instead of two lines you can initialize both variables at once.